jwst/ami/lg_model.py

```diff
@@ -388,14 +388,14 @@ def fit_image(
             )
         else:
             self.fittingmodel = modelin
-        if self.weighted is False:
-            self.soln, self.residual, self.cond, self.linfit_result = (
-                leastsqnrm.matrix_operations(image, self.fittingmodel, dqm=dqm)
-            )
-        else:
+        if self.weighted:
```

Comment:
I changed this check as it didn't quite match the log message about the kwarg in `FringeFitter`. The log message reported `Poisson` if `weighted is True` whereas this code tests `weighted is False`. If someone were to provide a non-bool value (1 for instance), the log would report equal weighting but poisson weighting would be used.